applications/tari_dan_wallet_daemon/src/services/account_monitor.rs

```diff
@@ -244,6 +244,10 @@ where
             )?;
         }
 
+        if !accounts_api.has_vault(&vault_addr)? {
```

Comment:
Shouldn't we add this check before adding the vault? around line 228

Reply:
yes